### PR TITLE
Adding Linear Algebra function to tensor flow fronted

### LINKED ISSUE #4533: cholesky
#1140

### PR DESCRIPTION
**Pull request by 14Emanuel**
Close #4533

### CODE REVIEW

**Comment by 14Emanuel** [comment]
Hello @simonetgordon , I have worked on the issue #4533 but every time I try to make a pull request #4557 the one of the checks keep on failing , can you assist me on this??

**Comment by 14Emanuel** [comment]
Hello @simonetgordon the checks logs indicate that my PR is failing because of a blank line in the PyTorch frontend file, but I am working on the TensorFlow frontend file and my code doesn't have that blank line at the bottom of the code, what might be wrong in this situation???

**Comment by kimbring2** [comment]
@14Emanuel Hello, thank you for your contribution. You can just ignore the error of other files for now. 

You need to modify the name of the argument of function same as the [official document of Tensorflow](https://www.tensorflow.org/api_docs/python/tf/linalg/cholesky). Otherwise, it gives an error.

Additionally, please refer to the [Cholesky test on the other frontend](https://github.com/unifyai/ivy/blob/d2e0f777962bcc93aa4c8ee75dd7b8bf22c8d728/ivy_tests/test_ivy/test_frontends/test_jax/test_jax_lax_linalg.py#L28) to check the proper way to generate the input matrix. That matrix should be a form of symmetric positive-definite.

**Comment by 14Emanuel** [comment]
Hello @kimbring2 , I have fixed the name of the argument. About the second problem I am a bit confused can you elaborate more????

**Comment by simonetgordon** [comment]
Hi @14Emanuel, you need to check the 'details' -> 'Run x tests' of the checks and then search for `cholesky` to see if it is causing any failures.

What @kimbring2 said is quite clear, the data generation for cholesky tensorflow (which you are working on) should only have symmetric  positive-definite matrices as input. This is done correctly for the jax cholesky unit test on [line 39](https://github.com/unifyai/ivy/blob/d2e0f777962bcc93aa4c8ee75dd7b8bf22c8d728/ivy_tests/test_ivy/test_frontends/test_jax/test_jax_lax_linalg.py#L39). More generally, you should compare the test in in the jax frontend with what you have written to make sure you're not missing anything.

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py [hunk 1]
```diff
@@ -394,3 +394,40 @@ def test_tensorflow_eye(
...
+        input_dtypes=input_dtype,
+        as_variable_flags=as_variable,
+        with_out=False,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        fw=fw,
+        frontend="tensorflow",
+        fn_tree="cholesky",
+        rtol=1e-02,
+        x=x,,
```
Comment (simonetgordon) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:432]: double comma here, delete one

**Review by simonetgordon — CHANGES_REQUESTED** [review]

**Comment by simonetgordon** [comment]
Hi @14Emanuel,

Could you merge with master so that updates to the test_frontends check can be seen? Currently it is not running due to now fixed issues on master.

Thanks!

Simone

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py (hunk 1 above)
Comment (14Emanuel) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:432]: Ok thank you, working on it right away

**Comment by 14Emanuel** [comment]
Hello, @simonetgordon I have fixed the syntax error and merged the main branch with my master.

**Comment by 14Emanuel** [comment]
Hello @simonetgordon I can see an error in this PR that the branch has some conflicts. I can clearly see in the logs that there may exist a problem between my master and main branch, I've tried fixing it by merging the main to the master branch but it still exists, can guide me on the exact problem so that I can work on it ASAP. 

Thanks
Emanuel

**Comment by simonetgordon** [comment]
@14Emanuel Hi there,

You should be able to see the issue by clicking 'Resolve conflicts' and seeing the difference between the master (Ivy) and main (your) branches. You want to match the Ivy master branch. If you're unsure about how to resolve github conflicts, please have a look at the many online resources there are for doing so, such as [this tutorial](https://www.atlassian.com/git/tutorials/using-branches/merge-conflicts).

**Review by simonetgordon — COMMENTED** [review]
@14Emanuel Hi there,

You should be able to see the issue by clicking 'Resolve conflicts' and seeing the difference between the master (Ivy) and main (your) branches. You want to match the Ivy master branch. If you're unsure about how to resolve github conflicts, please have a look at the many online resources there are for doing so, such as [this tutorial](https://www.atlassian.com/git/tutorials/using-branches/merge-conflicts).

**Comment by 14Emanuel** [comment]
Hello,  @simonetgordon in my process of trying to fix the merge conflict and I don't know which button I accidentally clicked in my VS Code to tamper with almost 336 files excluding mine, is there a way to reverse this mistake or solve the problem?

**Comment by 14Emanuel** [comment]
Hello, @simonetgordon I have fixed the file that had a merge conflict. But now I have another problem in my Pull Request, there are almost 336 changes that I did not personally change and I don't how to resolve them. Help out. Thank you.

**Review by simonetgordon — COMMENTED** [review]
I'm not really sure what you've done. But it seems like you may have checked out an older version of the source code instead of merging with the most recent changes.

Continue to research more on using git to try and understand how to fix your fork. Otherwise, there might be a 'sync with master' option on your fork on the github page which will merge the most recent changes to master into your fork, but will also delete the changes you've made to the tensorflow frontend. That means you'll have to re-implement the changes.

**Comment by 14Emanuel** [comment]
Hello @simonetgordon, I think implementing the features afresh will be easier than researching the problem because I have spent the last 8 hours trying to fix it and I am still not getting a clue about the problem. does implementing mean I can create a new PR for the same? If yes give me the directive, I am ready to implement the open source afresh since that will be easier and quicker for me. Thanks

**Comment by 14Emanuel** [comment]
Hello, @simonetgordon I managed to fix the problem by deleting all commits and committing the codes again. Can you request the @ivy-seed to remove some of the labels that came with my first mistake (the JAX, NumPy frontends, etc.) that are causing my checks to fail? Thank you.

File: ivy/functional/frontends/tensorflow/linalg.py [hunk 2]
```diff
@@ -43,3 +43,10 @@ def eye(num_rows, num_columns=None, batch_shape=None, dtype=ivy.float32, name=No
 
 
 eye.unsupported_dtypes = ("float16", "bfloat16")
+
+
+eye.unsupported_dtypes = ("float16", "bfloat16")
```
Comment (simonetgordon) [inline ivy/functional/frontends/tensorflow/linalg.py:48]: repeated code on lines 48 and 45

**Review by simonetgordon — CHANGES_REQUESTED** [review]

**Comment by simonetgordon** [comment]
Hi @14Emanuel, don't worry about the other test failures. Just the test-frontend-tensorflow and lint tests are important. There are still errors in the test-frontend-tensorflow related to cholesky. Are you running the test function locally to debug? See [here]( https://lets-unify.ai/ivy/contributing/0_setting_up.html?highlight=setting+up+test#setting-up-testing) for instructions on how to run tests on PyCharm and VSCode.
